From 5f3cc5a970d6710af7b4f4650a32f4ed0e817076 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Enrique=20Oca=C3=B1a=20Gonz=C3=A1lez?= <eocanha@igalia.com>
Date: Wed, 25 May 2016 13:38:01 +0000
Subject: [PATCH] qtdemux: Don't reset pending segment when running in time and
 mss mode

This keeps using the segment with the right start time received in the
demuxer sink pad, instead of resetting it to start=0.

https://bugzilla.gnome.org/show_bug.cgi?id=766868
---
 gst/isomp4/qtdemux.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/gst/isomp4/qtdemux.c b/gst/isomp4/qtdemux.c
index d21da31..1bf03a0 100644
--- a/gst/isomp4/qtdemux.c
+++ b/gst/isomp4/qtdemux.c
@@ -2108,8 +2108,10 @@ gst_qtdemux_handle_sink_event (GstPad * sinkpad, GstObject * parent,
       GST_DEBUG_OBJECT (demux, "Pushing newseg %" GST_SEGMENT_FORMAT, &segment);
       segment_event = gst_event_new_segment (&segment);
       gst_event_set_seqnum (segment_event, gst_event_get_seqnum (event));
-      /* erase any previously set segment */
-      gst_event_replace (&demux->pending_newsegment, NULL);
+      if (!(demux->mss_mode && demux->upstream_format_is_time)) {
+        /* erase any previously set segment */
+        gst_event_replace (&demux->pending_newsegment, NULL);
+      }
       gst_qtdemux_push_event (demux, segment_event);
 
       /* clear leftover in current segment, if any */
-- 
1.9.1

